rdock-utils/rdock_utils/sdrmsd/sdrmsd.py

```diff
@@ -0,0 +1,173 @@
+import functools
+import logging
+import math
+
+from openbabel import pybel
+
+from .superpose3d import MolAlignmentData, Superpose3D
+from .types import AutomorphismRMSD, CoordsArray, PoseMatchData, SDRMSDData
+
+logger = logging.getLogger("SDRMSD")
+
+
+class SDRMSD:
+    def __init__(
+        self,
+        reference_sdf: str,
+        input_sdf: str,
+        fit: bool,
+        threshold: float,
+        output_filename: str,
```

Comment:
I think the names for the arguments (and members) should be a bit clearer.
For me it makes more sense to have all the filenames together, and set default values for the other parameters.
threshold should be an optional float with default value of None (so the check below would be `if self.threshold is not None` instead of the plain `if self.threshold`, see related comment)
```suggestion
        reference_filename: str,
        input_filename: str,
        output_filename: str,
        fit: bool = False,
        threshold: float | None = None,

```

Reply:
I've made some changes. Please review them and let me know if I missed anything :+1:

Reply:
:100: :rocket: